### PR TITLE
chore: Refactor code quality issues

### PR DESCRIPTION
**Pull request by ankitdobhal**
## Description
Hey 👋 , I'm member of the Developer Outreach team at DeepSource and ran DeepSource analysis on my fork of the repo. It found some interesting [code quality improvements](https://deepsource.io/gh/ankitdobhal/tidb/issues) to consider.

This PR fixes a few of the issues detected for you to assess if it is right for you.
Happy to provide the tweaks separately otherwise :)

### What problem does this PR solve?

This PR improve some code quality issues like anti-pattern and bug risks. 

### What is changed and how it works?

- Remove unnecessary use of slice 
- Fix check for empty string
- Remove unnecessary wrapping of function call 
- Added .deepsource.toml file for continuous analysis on bug risks/performance/code-quality issues on new changes.

### Check List 

Tests 

- Unit test
- Integration test

### Release note 

- No release note

### CODE REVIEW

**Comment by aierui** [comment]
mark

File: .deepsource.toml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+version = 1
```
Comment (xhebox) [inline .deepsource.toml:1]: I think we should exclude this file.

**Review by xhebox — COMMENTED** [review]
Rest LGTM

File: .deepsource.toml (hunk 1 above)
Comment (ankitdobhal) [reply .deepsource.toml:1]: @xhebox should I remove the `.toml` file ?

Comment (xhebox) [reply .deepsource.toml:1]: I personally prefer to remove. And I think PR will get merged easier if it is removed. This file may be controversial.

@andylokandy  I've shared deepsource to devs. But there is no discussion yet.

Comment (ankitdobhal) [reply .deepsource.toml]: @xhebox removed `.toml` file.

**Comment by ankitdobhal** [comment]
Hey @xhebox , @tangenta  I have opened up issue #23827 regarding to integrate deepsource into CI.
Kindly have a look into issue.
Thanks and regards.

**Pull request merged**